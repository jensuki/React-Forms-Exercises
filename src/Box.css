.Box {

    background-color: lightcoral;
    margin-right: 10px;
    position: relative;
}

.Box-Removebtn {
    position: absolute;
    top: 0px;
    right: -45px;
    background-color: white;
    border: 1px solid black;
    cursor: pointer;
}