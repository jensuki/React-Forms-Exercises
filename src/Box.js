import React from 'react';
import './Box.css';

const Box = ({ id, width, height, backgroundColor, removeBox }) => {
    const handleRemove = () => {
        removeBox(id);
    }

    return (
        <div
            className="Box"
            data-testid="box-div"
            style={{
                width: `${width}px`,
                height: `${height}px`,
                backgroundColor: backgroundColor,
                margin: '10px'
            }}>
            <button className="Box-Removebtn" onClick={handleRemove}>X</button>
        </div>
    )
}

export default Box;