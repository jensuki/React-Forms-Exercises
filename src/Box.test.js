import { render, screen } from '@testing-library/react';
import Box from './Box';

it('renders without crashing', () => {
    render(
        <Box
            id="1"
            width={100}
            height={100}
            backgroundColor="blue"
            removeBox={() => { }}
        />
    )
})
it('matches snapshot', () => {
    const { asFragment } = render(
        <Box
            id="1"
            width={100}
            height={100}
            backgroundColor="blue"
            removeBox={() => { }}
        />
    );
    expect(asFragment()).toMatchSnapshot();
})


it('applies the correct styles', () => {
    render(
        <Box
            id="1"
            width={100}
            height={100}
            backgroundColor="blue"
            removeBox={() => { }}
        />
    );

    const boxDiv = screen.getByTestId('box-div');

    expect(boxDiv).toHaveStyle({
        width: '100px',
        height: '100px',
        backgroundColor: 'blue'
    })
});