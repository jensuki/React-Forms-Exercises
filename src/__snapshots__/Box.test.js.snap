// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    data-testid="box-div"
    style="width: 100px; height: 100px; background-color: blue; margin: 10px;"
  >
    <button
      class="Box-Removebtn"
    >
      X
    </button>
  </div>
</DocumentFragment>
`;
